onsp_co2_purchase/migrations/16.0.1.1.0/pre-create_fill_columns.py

```diff
@@ -0,0 +1,27 @@
+def migrate(cr, version):
+    if not version:
+        return
+
+    cr.execute(
+        """
+        ALTER TABLE purchase_order_line
+        ADD COLUMN carbon_origin_name character varying,
+        ADD COLUMN carbon_origin_value character varying;
+        """
+    )
+
+    queries = []
+    cr.execute("SELECT id, carbon_value_origin FROM purchase_order_line WHERE carbon_value_origin IS NOT NULL AND carbon_value_origin!='';")
+    for row in cr.fetchall():
+        origin_vals = row[1].split("|")
+        if len(origin_vals) == 2:
+            name = origin_vals[0]
+            value = origin_vals[1]
+        else:
+            name = row[1]
+            value = '0.0'
+
+        queries.append(f"UPDATE purchase_order_line SET carbon_origin_name='{name}', carbon_origin_value='{value}' WHERE id={row[0]}")
```

Comment:
As for in onsp_co2, i think we could do it in the first execute with this:

UPDATE purchase_order_line 
SET
    carbon_origin_name = CASE 
        WHEN POSITION('|' IN carbon_value_origin) > 0 
        THEN LEFT(carbon_value_origin, POSITION('|' IN carbon_value_origin) - 1)
        ELSE carbon_value_origin 
    END,
    carbon_origin_value = CASE 
        WHEN POSITION('|' IN carbon_value_origin) > 0 
        THEN RIGHT(carbon_value_origin, LENGTH(carbon_value_origin) - POSITION('|' IN carbon_value_origin))
        ELSE '0.0' 
    END
WHERE
    carbon_value_origin IS NOT NULL 
    AND carbon_value_origin != '';